packages/components/app/styles/components/side-nav/main.scss

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// SIDE-NAV
+//
+
+.hds-side-nav {
+  position: sticky;
+  top: 0;
+  z-index: 20; // needs to stay above the main content
+  height: 100vh;
+  min-height: 100vh;
+  isolation: isolate; // used to create a new stacking context (so we can set the overlay's z-index to -1)
+
+  // RESPONSIVENESS - This controls the width of the top-level container ("sidebar") in the grid, and impacts the available space for the "main" grid area
+
+  &.hds-side-nav--is-responsive {
+    transition:
+      width var(--hds-app-sidenav-animation-duration)
+      var(--hds-app-sidenav-animation-easing);
+  }
+
+  // mobile
+  &.hds-side-nav--is-mobile {
+    width: var(--hds-app-sidenav-width-minimized);
+  }
+
+  // desktop
+  &.hds-side-nav--is-desktop {
+    width: var(--hds-app-sidenav-width-expanded);
+  }
+}
+
+
+// OVERLAY
+
+.hds-side-nav__overlay {
+  position: fixed;
+  z-index: -1;
+  inset: 0;
+  background-color: var(--token-color-palette-neutral-700);
+  opacity: 0.2;
+  transition:
+    opacity var(--hds-app-sidenav-animation-duration)
+    var(--hds-app-sidenav-animation-easing)
+    var(--hds-app-sidenav-animation-delay);
+
+  // when we're minimized (mobile) we don't want the overlay to be visible/interactive
+  .hds-side-nav--is-minimized & {
+    opacity: 0;
+    pointer-events: none;
+  }
+
+  // when it's desktop we _never_ want the overlay to be visible
+  .hds-side-nav--is-desktop & {
+    display: none;
+  }
+}
+
+
+// MENU/TOGGLE BUTTON
+
+.hds-side-nav__menu-toggle-button {
+  position: absolute;
+  z-index: 1;
+  color: var(--token-color-foreground-high-contrast);
+  background: none;
+  border: 1px solid transparent;
+  cursor: pointer;
+  transition-timing-function: var(--hds-app-sidenav-animation-easing);
+  transition-duration: var(--hds-app-sidenav-animation-duration);
+  transition-property: width, height, padding, border-radius, transform;
+
+  &:hover {
+    background-color: var(--token-color-palette-neutral-600);
+    border-color: transparent;
+  }
+
+  &:active {
+    background-color: var(--token-color-palette-neutral-500);
+    border-color: transparent;
+  }
+
+  &:focus {
+    background-color: var(--token-color-palette-neutral-500);
+    border-color: var(--token-color-focus-action-internal);
+    outline: 3px solid var(--token-color-focus-action-external);
+  }
+
+  // by default it's styled as "close" button
+  .hds-side-nav--is-mobile & {
+    width: 24px;
+    height: 24px;
+    padding: 1px;
+    background-color: var(--token-color-foreground-primary);
+    border-radius: 3px;
+    transform:
+      translateX(calc(var(--hds-app-sidenav-width-expanded) + 8px))
```

Comment:
What is the 8px for?

Reply:
shift from the border
<img width="445" alt="modified" src="https://user-images.githubusercontent.com/686239/233321068-89cd8d72-0f26-4356-abab-ea52b6959ef4.png">